src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveExtensions.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using System.Security.Cryptography;
+
+namespace Xamarin.Android.Tasks
+{
+	public static class ZipArchiveExtensions 
+	{
+		static SHA1 sha1 = SHA1.Create ();
```

Comment:
This should *not* be static, because [instance members are not guaranteed to be thread safe](https://msdn.microsoft.com/en-us/library/system.security.cryptography.sha1(v=vs.110).aspx).

Additionally, the type implements `IDisposable`, so presumably use should be wrapped in `using` blocks.